config/configuration.rb

```diff
@@ -1,23 +1,115 @@
+require 'pathname'
+require 'dotenv'
+
 # dashboard app specific configuration
 class Configuration
-  class << self
-    attr_writer :app_development_enabled
-    attr_writer :app_sharing_enabled
-
-    # FIXME: temporary
-    attr_accessor :app_sharing_facls_enabled
-    alias_method :app_sharing_facls_enabled?, :app_sharing_facls_enabled
-
-    def app_development_enabled?
-      return @app_development_enabled if defined? @app_development_enabled
-      ENV['OOD_APP_DEVELOPMENT'].present? || DevRouter.base_path.exist?
-    end
-    alias_method :app_development_enabled, :app_development_enabled?
-
-    def app_sharing_enabled?
-      return @app_sharing_enabled if defined? @app_sharing_enabled
-      @app_sharing_enabled = ENV['OOD_APP_SHARING'].present?
-    end
-    alias_method :app_sharing_enabled, :app_sharing_enabled?
+  attr_writer :app_development_enabled
+  attr_writer :app_sharing_enabled
+
+  # FIXME: temporary
+  attr_accessor :app_sharing_facls_enabled
+  alias_method :app_sharing_facls_enabled?, :app_sharing_facls_enabled
+
+  # The app's configuration root directory
+  # @return [Pathname] path to configuration root
+  def config_root
+    Pathname.new(ENV["OOD_APP_CONFIG_ROOT"] || "/etc/ood/config/apps/dashboard")
+  end
+
+  def initializers_root
+    Pathname.new(ENV["OOD_APP_INITIALIZERS_ROOT"] || config_root.join("initializers"))
+  end
+
+  def load_external_config?
+    to_bool(ENV.fetch('OOD_LOAD_EXTERNAL_CONFIG', (rails_env == 'production')))
+  end
+
+  # Load the dotenv local files first, then the /etc dotenv files and
+  # the .env and .env.production or .env.development files.
+  #
+  # Doing this in two separate loads means OOD_APP_CONFIG_ROOT can be specified in
+  # the .env.local file, which will specify where to look for the /etc dotenv
+  # files. The default for OOD_APP_CONFIG_ROOT is /etc/ood/config/apps/myjobs and
+  # both .env and .env.production will be searched for there.
+  def load_dotenv_files
+    # .env.local first, so it can override OOD_APP_CONFIG_ROOT
+    Dotenv.load(*dotenv_local_files) unless dotenv_local_files.empty?
+
+    # load the rest of the dotenv files
+    Dotenv.load(*dotenv_files)
+  end
+
+  def app_development_enabled?
+    return @app_development_enabled if defined? @app_development_enabled
+    to_bool(ENV.fetch('OOD_APP_DEVELOPMENT', DevRouter.base_path.exist?))
+  end
+  alias_method :app_development_enabled, :app_development_enabled?
+
+  def app_sharing_enabled?
+    return @app_sharing_enabled if defined? @app_sharing_enabled
+    @app_sharing_enabled = to_bool(ENV['OOD_APP_SHARING'])
```

Comment:
The assignment `@app_sharing_enabled = ...` isn't necessary here. See the above `app_development_enabled?` method.

Reply:
If you feel it is necessary, then you may need to add the assignment to the method `app_development_enabled?`.